pkgs/os-specific/bsd/freebsd/by-name/mkDerivation.nix

```diff
@@ -0,0 +1,60 @@
+{ lib, crossLibcStdenv, stdenv, hostVersion, buildPackages, buildFreebsd, hostArchBsd, compatIfNeeded, filterSource, ... }:
+lib.makeOverridable (attrs: let
+  # the use of crossLibcStdenv in the isStatic case is kind of a misnomer but I think it works
+  stdenv' = if (attrs.isStatic or false) then crossLibcStdenv else stdenv;
+in stdenv'.mkDerivation (rec {
+  pname = "${attrs.pname or (baseNameOf attrs.path)}";
+  version = hostVersion;
+  src = filterSource { inherit pname; inherit (attrs) path; extraPaths = attrs.extraPaths or []; };
+
+  nativeBuildInputs = [
+    buildPackages.bsdSetupHook buildFreebsd.freebsdSetupHook
+    buildFreebsd.bmakeMinimal  # TODO bmake??
+    buildFreebsd.install buildFreebsd.tsort buildFreebsd.lorder buildPackages.mandoc buildPackages.groff #statHook
+  ] ++ attrs.extraNativeBuildInputs or [];
+  buildInputs = compatIfNeeded;
+
+  HOST_SH = stdenv'.shell;
+
+  # Since STRIP below is the flag
+  STRIPBIN = "${stdenv'.cc.bintools.targetPrefix}strip";
+
+  makeFlags = [
+    "STRIP=-s" # flag to install, not command
+  ] ++ lib.optional (!stdenv'.hostPlatform.isFreeBSD) "MK_WERROR=no";
+
+  # amd64 not x86_64 for this on unlike NetBSD
+  MACHINE_ARCH = hostArchBsd;
+
+  MACHINE = hostArchBsd;
+
+  MACHINE_CPUARCH = MACHINE_ARCH;
+
+  COMPONENT_PATH = attrs.path or null;
+
+  strictDeps = true;
+
+  meta = with lib; {
+    maintainers = with maintainers; [ rhelmot artemist ];
+    platforms = platforms.unix;
+    license = licenses.bsd2;
+  } // attrs.meta or {};
+} // lib.optionalAttrs stdenv'.hasCC {
+  # TODO should CC wrapper set this?
+  CPP = "${stdenv'.cc.targetPrefix}cpp";
+} // lib.optionalAttrs stdenv'.isDarwin {
+  MKRELRO = "no";
+} // lib.optionalAttrs (stdenv'.cc.isClang or false) {
+  HAVE_LLVM = lib.versions.major (lib.getVersion stdenv'.cc.cc);
+} // lib.optionalAttrs (stdenv'.cc.isGNU or false) {
+  HAVE_GCC = lib.versions.major (lib.getVersion stdenv'.cc.cc);
+} // lib.optionalAttrs (stdenv'.isx86_32) {
+  USE_SSP = "no";
```

Comment:
We could simplify those conditions by using env.

Reply:
This code actually predates me - see https://github.com/NixOS/nixpkgs/blob/a7d38622e750d08f14de9c671caf88976b54ad00/pkgs/os-specific/bsd/freebsd/default.nix#L139-L153

Do you still want it changed?